sequential/dpm.py

```diff
@@ -0,0 +1,146 @@
+        Minimum frequency threshold for negative sequences. min_frequency_neg = 0.3 by default.
+        If int, represents the minimum number of sequences (rows) a pattern should occur
+        If float, should be (0.0, 1.0] and represents
+        the minimum percentage of sequences (rows) a pattern should occur.
+
+    Returns
+    -------
+    Dictionary of DPM patterns with keys for union, intersection, unique_pos and unique_neg.
+    Each aggregation has the patterns to be sorted.
+
+    """
+
+    # Mine positive cohort
+    patterns_pos = seq2pat_pos.get_patterns(min_frequency=min_frequency_pos)
+
+    # Mine negative cohort
+    patterns_neg = seq2pat_neg.get_patterns(min_frequency=min_frequency_neg)
+
+    # Drop frequencies in the end of mined patterns
+    patterns_pos = drop_frequency(patterns_pos)
+    patterns_neg = drop_frequency(patterns_neg)
+
+    aggregation_to_patterns = dict()
+
+    intersection_patterns = set(map(tuple, patterns_pos)).intersection(set(map(tuple, patterns_neg)))
+    aggregation_to_patterns[DichotomicAggregation.intersection] = sorted(list(map(list, intersection_patterns)))
+
+    union_patterns = set(map(tuple, patterns_pos)).union(set(map(tuple, patterns_neg)))
+    aggregation_to_patterns[DichotomicAggregation.union] = sorted(list(map(list, union_patterns)))
+
+    unique_neg_patterns = set(map(tuple, patterns_neg)) - set(map(tuple, patterns_pos))
+    aggregation_to_patterns[DichotomicAggregation.unique_neg] = sorted(list(map(list, unique_neg_patterns)))
+
+    unique_pos_patterns = set(map(tuple, patterns_pos)) - set(map(tuple, patterns_neg))
+    aggregation_to_patterns[DichotomicAggregation.unique_pos] = sorted(list(map(list, unique_pos_patterns)))
+
+    return aggregation_to_patterns
+
+
+def get_one_hot_encodings(sequences: List[list], patterns: List[list],
+                          constraints: Union[List[_Constraint], None] = None,
+                          rolling_window_size: int = 10, drop_pattern_frequency=False) -> pd.DataFrame:
+    """
+    Create a data frame having one-hot encoding of sequences.
+
+    Parameters
+    ----------
+    sequences: List[list]
+        A list of sequences of items.
+    patterns: List[list]
+        A list of interested patterns, which defines the encoding space.
+    constraints: Union[list, None]
+        The constraints enforced in the creation of encoding
+    rolling_window_size: int
+        The rolling window along a sequence within which patterns are detected locally. It controls the length of
+        sequence subject to the pattern detection, to speed up the encodings generation.
+        (rolling_window_size=10 by default).
+    drop_pattern_frequency: bool
+        Drop the frequency appended in the end of each input pattern, drop_pattern_frequency=False by default.
+
+    Returns
+    -------
+    A data frame having one-hot encoding of sequences using the given patterns.
+    
+    Example
+        sequence      feat0 feat1 feat2 ...
```

Comment:
Minor comment: I am not sure about the "feat" columns.. Why not spell this out completely, feature, and append an underscore so that from the outside someone can split easily if needed (e..g, grab some subset) 

So I would be explicit here and call this "feature_0", "feature_1" and so on

Reply:
Agree. `feature_` makes it clearer.